src/Serialization/WitnessSet.purs

```diff
@@ -2,23 +2,42 @@ module Serialization.WitnessSet where
 
 import Prelude
 
+import Data.Array as Array
 import Data.Maybe (maybe)
 import Data.Newtype (unwrap)
 import Data.Traversable (for_, traverse, traverse_)
 import Data.Tuple.Nested ((/\))
+import Deserialization.FromBytes (fromBytesEffect)
 import Effect (Effect)
 import Effect.Exception (throw)
-import Types.ByteArray (ByteArray)
-
-import Serialization.Types (BigNum, BootstrapWitness, Ed25519Signature, ExUnits, NativeScripts, PlutusData, PlutusScript, PlutusScripts, PublicKey, Redeemer, RedeemerTag, TransactionWitnessSet, Vkey, Vkeywitness, Vkeywitnesses)
-import Deserialization.FromBytes (fromBytesEffect)
 import FfiHelpers (ContainerHelper, containerHelper)
 import Serialization.BigNum (bigNumFromBigInt)
 import Serialization.NativeScript (convertNativeScripts)
-import Types.Transaction as T
+import Serialization.Types
+  ( BigNum
+  , BootstrapWitness
+  , Ed25519Signature
+  , ExUnits
+  , NativeScripts
+  , PlutusData
+  , PlutusScript
+  , PlutusScripts
+  , PublicKey
+  , Redeemer
+  , RedeemerTag
+  , TransactionWitnessSet
+  , Vkey
+  , Vkeywitness
+  , Vkeywitnesses
+  )
 import Types.Aliases (Bech32String)
-
+import Types.ByteArray (ByteArray)
 import Types.RedeemerTag as Tag
+import Types.Transaction as T
+
+setPlutusData :: PlutusData -> TransactionWitnessSet -> Effect TransactionWitnessSet
+setPlutusData pd ws = _wsSetPlutusData containerHelper ws (Array.singleton pd)
+  *> pure ws
```

Comment:
Why an argument is returned here?

Reply:
I think that not returning mutable values out of functions mutating them highlights that the values are mutable references of values, not pure data. If the caller made this call, they must have the witness set in scope, so no need to return.

Reply:
Fair enough, I'll change that